pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go

```diff
@@ -28,7 +29,10 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-var kubeconfigWriterImage = flag.String("kubeconfig-writer-image", "override-with-kubeconfig-writer:latest", "The container image containing our kubeconfig writer binary.")
+var (
+	kubeconfigWriterImage = flag.String("kubeconfig-writer-image", "override-with-kubeconfig-writer:latest", "The container image containing our kubeconfig writer binary.")
+	bashNoopImage         = flag.String("bash-noop-image", "override-with-bash-noop:latest", "The container image containing bash shell")
```

Comment:
before I get too far into this PR - could you explain a bit (maybe even in some comments in the `main.go`s?) a bit more about what these binaries are for? it's a bit hard to tell at a glance

do we _need_ a generic bash image? alternatively we could run any other container that has bash in it, maybe set the entrypoint to `bash`?

Reply:
> do we need a generic bash image?
I do not think so but I remember in `build` there was an image `build-base` which was custom to what build base image required (I think). I am not entirely sure why we need new custom image in this case because `.ko.yaml` lets you perfectly override that if user wants different base image. I added comment in `.ko.yaml` file about the expectation of image.